### PR TITLE
[Enhancement] load delvec when tablet scanner is openned

### LINKED ISSUE #12721: [Enhancement] load delvec when tablet scanner is openned
## Enhancement
To prevent delvec not found error when query runs for a very long time(>30min) and delvec got GCed.

### PR DESCRIPTION
**Pull request by decster**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #12721

## Problem Summary(Required) ：

This PR changes segment_iterator, get delvec when segment_iterator is created, not in `_init`, so delvecs are got when the query started, preventing delvec not found error when query runs for a very long time(>30min) and delvec got GCed.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Review by trueeyu — DISMISSED** [review]

File: be/src/storage/rowset/segment_iterator.cpp [hunk 1]
```diff
@@ -307,19 +308,30 @@ SegmentIterator::SegmentIterator(std::shared_ptr<Segment> segment, vectorized::S
...
         TabletSegmentId tsid;
         tsid.tablet_id = _opts.tablet_id;
         tsid.segment_id = _opts.rowset_id + segment_id();
-        RETURN_IF_ERROR(
-                StorageEngine::instance()->update_manager()->get_del_vec(_opts.meta, tsid, _opts.version, &_del_vec));
-        if (_del_vec && _del_vec->empty()) {
-            _del_vec.reset();
+        _get_del_vec_st =
+                StorageEngine::instance()->update_manager()->get_del_vec(_opts.meta, tsid, _opts.version, &_del_vec);
+        if (_get_del_vec_st.ok()) {
```
Comment (trueeyu) [inline be/src/storage/rowset/segment_iterator.cpp:321]: if _get_del_vect_st is not ok, what will happened?

File: be/src/exec/pipeline/scan/olap_chunk_source.cpp [hunk 2]
```diff
@@ -84,6 +84,9 @@ void OlapChunkSource::_init_counter(RuntimeState* state) {
     _cached_pages_num_counter = ADD_COUNTER(_runtime_profile, "CachedPagesNum", TUnit::UNIT);
     _pushdown_predicates_counter = ADD_COUNTER(_runtime_profile, "PushdownPredicates", TUnit::UNIT);
 
+    _get_rowset_timer = ADD_TIMER(_runtime_profile, "GetRowsets");
```
Comment (trueeyu) [inline be/src/exec/pipeline/scan/olap_chunk_source.cpp:87]: _get_rowsets_timer?

File: be/src/storage/rowset/segment_iterator.cpp (hunk 1 above)
Comment (trueeyu) [reply be/src/storage/rowset/segment_iterator.cpp:321]: return err msg in `_init`?

Comment (decster) [reply be/src/storage/rowset/segment_iterator.cpp:321]: the following get_next will return an error, same as before.

File: be/src/exec/pipeline/scan/olap_chunk_source.cpp (hunk 2 above)
Comment (decster) [reply be/src/exec/pipeline/scan/olap_chunk_source.cpp]: ok

File: be/src/storage/rowset/segment_iterator.cpp (hunk 1 above)
Comment (decster) [reply be/src/storage/rowset/segment_iterator.cpp:321]: > return err msg in `_init`?

yes

**Review by trueeyu — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by decster** [comment]
@mergify backport branch-2.5

**Pull request merged**